### PR TITLE
Move psychologist immunity from mind to skillchip

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
tweak: Psychologist immunity to supermatter hallucinations, and the
ability to make it "calmer", is now provided by a skillchip, rather than
innate.
/:cl:

Skillchips are neat, and this is the sort of "job-specific skill" that
can remain job specific, but still gainable in the shift through
in-universe means.

The logic behind the skillchip is that the supermatter is psychically
sensitive, hence the hallucinations and headaches from looking at it,
and projecting the "correct mental aura" will help it calm down. But
that is lore following mechanics.

- Refactors skillchips to be able to automatically apply more than one
  trait.

### CODE REVIEW

File: code/modules/library/skill_learning/skillchip.dm [hunk 1]
```diff
@@ -7,8 +7,8 @@
 	custom_price = PAYCHECK_MEDIUM * 3
 	w_class = WEIGHT_CLASS_SMALL
 
-	/// Trait automatically granted by this chip, optional
-	var/auto_trait
+	/// Traits automatically granted by this chip, optional
+	var/list/auto_traits
```
Comment (interestingusernam3) [inline code/modules/library/skill_learning/skillchip.dm:11]: Replacement of auto_trait with auto_traits should probably be mentioned in the PR description.

File: code/modules/library/skill_learning/skillchip.dm [hunk 2]
```diff
@@ -142,8 +142,9 @@
 	if(!silent && activate_message)
 		to_chat(user, activate_message)
 
-	if(auto_trait)
-		ADD_TRAIT(user, auto_trait, SKILLCHIP_TRAIT)
+	if(auto_traits)
+		for(var/trait in auto_traits)
+			ADD_TRAIT(user, trait, SKILLCHIP_TRAIT)
```
Comment (Rohesie) [inline code/modules/library/skill_learning/skillchip.dm:147]:
```suggestion
	for(var/trait in auto_traits)
		ADD_TRAIT(user, trait, SKILLCHIP_TRAIT)
```
No need to check if the list exists.

File: code/modules/library/skill_learning/skillchip.dm [hunk 3]
```diff
@@ -176,8 +177,9 @@
 	if(!silent && deactivate_message)
 		to_chat(user, deactivate_message)
 
-	if(auto_trait)
-		REMOVE_TRAIT(user, auto_trait, SKILLCHIP_TRAIT)
+	if(auto_traits)
+		for(var/trait in auto_traits)
+			REMOVE_TRAIT(user, trait, SKILLCHIP_TRAIT)
```
Comment (Rohesie) [inline code/modules/library/skill_learning/skillchip.dm:182]:
```suggestion
	for(var/trait in auto_traits)
		REMOVE_TRAIT(user, trait, SKILLCHIP_TRAIT)
```

File: code/modules/library/skill_learning/skillchip.dm [hunk 4]
```diff
@@ -7,8 +7,8 @@
 	custom_price = PAYCHECK_MEDIUM * 3
 	w_class = WEIGHT_CLASS_SMALL
 
-	/// Trait automatically granted by this chip, optional
-	var/auto_trait
+	/// Traits automatically granted by this chip, optional
```
Comment (Rohesie) [inline code/modules/library/skill_learning/skillchip.dm:10]:
```suggestion
	/// Traits automatically granted by this chip, optional. Lazy list.
```

**Pull request merged**